llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -966,8 +984,9 @@ bool InferAddressSpacesImpl::updateAddressSpace(
   // of all its pointer operands.
   unsigned NewAS = UninitializedAddressSpace;
 
-  const Operator &Op = cast<Operator>(V);
-  if (Op.getOpcode() == Instruction::Select) {
+  if (isa<Operator>(V) &&
+      cast<Operator>(V).getOpcode() == Instruction::Select) {
+    const Operator &Op = cast<Operator>(V);
```

Comment:
The old code would crash on non-operator `V`. The new would carry on into the `else` branch.

I assume it's intentional, so we can process arguments, but I worry if we may be potentially allowing other unwanted types to sneak through. An assert on what we expect `V` to be would be prudent here.